File: java-1/src/test/java/day00/Day00Test.java
package day00;

import org.junit.jupiter.api.Test;
import utils.Utils;

import static org.assertj.core.api.Assertions.assertThat;

class Day00Test {

    @Test
    void solveFirstPuzzle() {
        var day00 = new Day00(Utils.readResource("samples/sample0.txt"));

        assertThat(day00.solvePuzzle1()).isEqualTo(23); // length of first line
    }

    @Test
    void solveSecondPuzzle() {
        var day00 = new Day00(Utils.readResource("samples/sample0.txt"));

        assertThat(day00.solvePuzzle2()).isEqualTo(2); // number of lines
    }

}

File: kotlin-1/src/test/kotlin/day00/Day00Test.kt
package day00

import io.kotest.matchers.shouldBe
import org.junit.jupiter.api.Test


class Day00Test {

    @Test
    internal fun `solve first puzzle with sample input`() {
        val input = sample.lineSequence()

        solvePuzzle1(input) shouldBe 23 // length of first line
    }

    @Test
    internal fun `solve second puzzle with sample input`() {
        val input = sample.lineSequence()

        solvePuzzle2(input) shouldBe 2 // number of lines
    }

    val sample = """
        This is a sample input.
        It has two lines.
    """.trimIndent()
}
